src/connector/src/source/kinesis/source/reader.rs

```diff
@@ -208,27 +216,37 @@ impl CommonSplitReader for KinesisSplitReader {
 }
 impl KinesisSplitReader {
     async fn new_shard_iter(&mut self) -> Result<()> {
-        let (starting_seq_num, iter_type) = if self.latest_offset.is_some() {
+        let (starting_seq_num, start_timestamp, iter_type) = if self.latest_offset.is_some() {
             (
                 self.latest_offset.clone(),
+                None,
                 ShardIteratorType::AfterSequenceNumber,
             )
         } else {
             match &self.start_position {
-                KinesisOffset::Earliest => (None, ShardIteratorType::TrimHorizon),
-                KinesisOffset::SequenceNumber(seq) => {
-                    (Some(seq.clone()), ShardIteratorType::AfterSequenceNumber)
-                }
-                KinesisOffset::Latest => (None, ShardIteratorType::Latest),
+                KinesisOffset::Earliest => (None, None, ShardIteratorType::TrimHorizon),
+                KinesisOffset::SequenceNumber(seq) => (
```

Comment:
I feel it would be better if we could give an warning to users, something like
"The sequence is only unique within a certain shard, so the semantics they expect may deviate from the reality, etc."


We give a warning instead of completely disabling it because we still want to keep backward compatibility

Reply:
I think it's a completely wrong design. Recommend removing it. (Yeah I know it's breaking change...)

Reply:
the warning here cannot propagate to the frontend. Let's make some breaking changes here.

Reply:
> the warning here cannot propagate to the frontend. Let's make some breaking changes here.

If possible, I suggest we can leave the option here to avoid `panic` (perhaps print a warning log), but remove all related implementation code.

Reply:
I think there is no `sequence_number` option, unless the shard is fixed and pass a `sequence_number` for each shard to rw. Otherwise, this feature doesn't make sense.